### DISCUSSION TITLE
Any risks in using coturn on Windows?

### DISCUSSION THREAD

**Discussion by irinelnistor** [coturn/coturn#1298] [Q&A] [answered]
Initially this library was created for linux and most instances are most likely running in linux. Do you think we are taking a great risk when considering to run coturn on Windows, taking into account future maintenance and the fact that probably new features will be added and tested primarily on linux. 
@KangLin you've done a great job on enabling it for windows. Are you using it in production on a windows environment, do you have any concerns?

**Comment by KangLin** [comment, marked as answer]
- So far, it is still being improved. Not recommended for production environments.
- From the perspective of ISSUS, running coturn on Windows has certain requirements. 
- Risk:
  - Operating system compatible
    1. Thread (pthread)
    2. Network (libevent)  
    There is already a solution. Just comply with using the cross-platform library in the solution. 
  - Stability testing on windows（TODO）
  - Performance testing on windows（TODO）

Welcome that interested friends will do it together.

**Reply by irinelnistor** [reply]
We managed to run it and make it work on Windows. And for our limited test it seemed stable, but we had very few connections

**Reply by irinelnistor** [reply]
@KangLin have you done any kind of test on it so far, are you planning to use it in a production environment in the future? Did you encounter any stability or performance issues? In our limited testing we didn't encounter substantial performance differences between running it on Linux or Windows.
We are not planning to go in production for at least a year, We will not have a great number of users  and we'll mostly use STUN with web rtc peer to peer connections, not TURN. Would such a scenario of using coturn on windows seem more feasible to you?

**Comment by KangLin** [comment]
Thank you for the test. 
Windows is still in development. I only did unit tests. I'm currently adding Windows services, and when all Windows features are developed, I'll do performance and stability testing.
Welcome to you are  do performance and stability tests together.














At 2023-10-23 14:22:27, "irinelnistor" ***@***.***> wrote:

@KangLin have you done any kind of test on it so far? Did you encounter any stability or performance issues? In our limited testing we didn't encounter substantial performance differences between running it on Linux or Windows.
